postgres_insert_block.py

```diff
@@ -0,0 +1,62 @@
+from .postgres_base_block import PostgresBase
+
+
+class PostgresInsert(PostgresBase):
+    """A block for inserting incoming signals into a postgres database.
+
+    Properties:
+
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def process_signals(self, signals):
+        """execute an insert command for all incoming signals"""
+        for signal in signals:
+            self.execute_insert(signal.to_dict())
+        # only commit transactions when all have been attempted
+        if self.commit_all():
+            self._commit_transactions()
+
+    def execute_insert(self, data):
+        """execute an insert query for the given data"""
+        self.logger.debug('executing INSERT on data: {}'.format(data))
+        try:
+            self._cur.execute(self._build_insert_query_string(data),
+                              tuple(data.values()))
+        except:
+            self.logger.exception("Could not execute command".format())
+            self._rollback_transactions()
```

Comment:
I don't think rolling back here makes sense. Imagine we send 3 signals in (and we need to commit). If the first signal worked, second did not work, and third worked, then only the third signal would be inserted (feels weird).

I think the logic that makes sense to me is:
- If we need to commit (commit transactions is `True`):
  - any exception in any incoming signal causes the whole transaction to be rolled back
  - implementation would look like [1] below
- If we don't need to commit (commit transactions is `False`):
  - any signal that raises an exception logs an error, but we continue
  - no rollback has to be called if we don't commit

Implementation [1] (add comments for implementation details, of course):
```python
if self._commit_transactions():
    try: 
        for signal in signals:
            self.execute_insert(signal)
        self._commit_transactions()
    except:
        self.logger.exception("uh oh")
        self._rollback_transactions()
else:
    for signal in signals:
        try:
            self.execute_insert(signal)
        except:
            self.logger.exception("uh oh")
```